### PR TITLE
Bug 1193576 - Add a setting to browse privately by default

### PR DESCRIPTION
**Pull request by jedireza**

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=bbc11409b54e47bf9226be1dda18db09c6a7b0a9

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=bdbc81c74962420e6424fc300c037cd1325b1ace

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=672fe0529e1dc1bec87ccdb65496f4018fae51f3

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=067eeb322e1bbe9ccfb3adf0285c9b67750daed0

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=a333002760926333af82cd6c77a65aadd74b5f2a

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=6fee84a6e7d7ec2031451942315f990974b34a01

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=630cb4263eafdd58e153d4d468d2f894e007cb94

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=1b8dd96a62013cde483e4c0a93a886712769aa2d

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=d9fbb6ea2c1f3ff720139b7c8f812382603dc0d7

**Comment by jedireza** [comment]
FYI: I'm getting linting errors when I use `new MozActivity(...`

```
apps/search/js/newtab.js: line 80, col 7, Do not use 'new' for side effects. (ERROR)
apps/search/js/search.js: line 331, col 7, Do not use 'new' for side effects. (ERROR)
```

And when I remove new my tests fail:

```
1) [search-test/unit/search_test.js] search/search submit Fires correct Event to metrics search for search term:
     TypeError: Constructor MozActivity requires 'new'
```

**Comment by KevinGrandon** [comment]
> FYI: I'm getting linting errors when I use new MozActivity(...

Yeah, we should fix this in the jshintrc file, but for now you can do something like adding the following to the top of the file:

```
/* jshint nonew: false */
```

File: apps/system/test/unit/browser_test.js [hunk 1]
```diff
@@ -81,4 +92,26 @@ suite('system/Browser', function() {
     var app = MockAppWindowHelper.mLatest;
     assert.equal(app.isPrivate, true);
   });
+
+  test('when private browsing is the default setting', function() {
+    MockNavigatorSettings.mTriggerObservers('browser.private.default',
+        {settingValue: true});
+
+    setTimeout(function () {
```
Comment (KevinGrandon) [inline apps/system/test/unit/browser_test.js]: Ok - so is this the problem? Instead of doing this, have you tried using sinon fake timers?

Comment (jedireza) [inline apps/system/test/unit/browser_test.js]: I have not yet, but did see this:

https://github.com/mozilla-b2g/gaia/blob/master/shared/test/unit/mocks/mock_navigator_moz_settings.js#L36-L37

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=6967804414b871037fff1417325335d8b3e02d26

**Comment by jedireza** [comment]
I've added `/* jshint nonew: false */` to the test files.

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=222f4072c97c2b0e5facd51e949d94121e24422a

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=2a4506bc5fc072c081ceef47a751f36fcfa75566

**Comment by jedireza** [comment]
@KevinGrandon the last commit demonstrates using sinon timers and `setImmediate` instead of `setTimeout`. Before using `setImmediate` I also tried sinon's `clock.tick(n)` with values of `30` - `2000`, with no luck.

FYI: `setImmediate` isn't available without enabling sinon timers.

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=95bd00e5c5ac2b3a40aa05ef6248dde602bb71c0

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=1e0a18db4be4631dfabd7faf3645152a203f1eaa

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=ac95ce94e7036129669a5b153ea49215d9ccf3b1

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=e22aeb2842a74e16444737e51db983facd062f6a

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=336a2a4e9c04a57df522f0346cd1986ec482273d

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=c468957aab854b76c4ca03f0a1870654cf50cbd8

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=c468957aab854b76c4ca03f0a1870654cf50cbd8

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=bea8e16079e37770db85468c7f851c79b559cc65

**Comment by jedireza** [comment]
Closing this PR for in favor of #31988

**Pull request closed without merging**